src/main/java/de/tum/in/www1/artemis/service/tutorialgroups/TutorialGroupService.java

```diff
@@ -612,4 +617,85 @@ private Optional<User> findStudent(StudentDTO studentDto, String studentCourseGr
         return userOptional.isPresent() && userOptional.get().getGroups().contains(studentCourseGroupName) ? userOptional : Optional.empty();
     }
 
+    /**
+     * Export all tutorial groups for a specific course and user to a CSV format.
+     *
+     * @param course The course for which the tutorial groups should be exported.
+     * @param user   The user for whom the transient properties of the tutorial groups should be set.
+     * @param fields The list of fields to include in the CSV export.
+     * @return A CSV string representing the tutorial groups for the given course and user.
+     * @throws IOException if an I/O error occurs during CSV generation.
+     */
+    public String exportTutorialGroupsToCSV(Course course, User user, List<String> fields) throws IOException {
+        Set<TutorialGroup> tutorialGroups = findAllForCourse(course, user);
+
+        StringWriter out = new StringWriter();
+        CSVFormat csvFormat = CSVFormat.DEFAULT.withHeader(fields.toArray(new String[0]));
+        try (CSVPrinter printer = new CSVPrinter(out, csvFormat)) {
```

Comment:
```suggestion
    try (CSVPrinter printer = new CSVPrinter(out, csvFormat)) {
    Map<String, Function<TutorialGroup, Object>> fieldActions = new HashMap<>();
    fieldActions.put("ID", TutorialGroup::getId);
    fieldActions.put("Title", TutorialGroup::getTitle);
    // ... add other fields here

    for (TutorialGroup tutorialGroup : tutorialGroups) {
        for (String field : fields) {
            Object value = Optional.ofNullable(fieldActions.get(field))
                                   .map(func -> func.apply(tutorialGroup))
                                   .orElse("");
            printer.print(value);
        }
        printer.println();
    }
}

```
This rewrite is more efficient since it uses a map and easier to add more fields in the future